edgelet/contrib/config/linux/template.toml

```diff
@@ -28,6 +28,21 @@
 # trust_bundle_cert = "file:///var/secrets/trust-bundle.pem"
 
 
+# ==============================================================================
+# Quickstart certificate expiration
```

Comment:
I think we should put this next to the Edge CA cert section.

Reply:
There is a limitation of our current TOML parser where individual elements need to precede the tables. So, individual properties need to be at the top.

Reply:
Put it *in* the `edge_ca` section? It only applies to the Edge CA after all.

Reply:
I went back and forth on this one, but ultimately I went with this approach because it is mutually exclusive of the other options in `edge_ca`. Putting it under `edge_ca` makes the auto-generation behavior seem related to the customer-specified cert, when its not. Previously, in 1.1, this section was called `certificates` where the trust-bundle, device CA and auto-generated workload CA properties were all bundled up in one. If `trust_bundle_cert` is split out, so should this.

Reply:
I think both Varun and Micah asked for `trust_bundle_cert` to be moved into `[edge_ca]` too actually, for exactly that reason.

Reply:
We should not conflate these two issues. Values that are mutually exclusive should not be co-located in the configuration. The config file clearly states that `edge_ca` is a customer-defined configuration, while `auto_generated_edge_ca_expiry_days` is a configuration, if there is no customer-defined `edge_ca`. The relationship between `trust_bundle` and `edge_ca` is different from the `auto_generated_edge_ca_expiry_days` and `edge_ca` relationship in that `edge_ca` value needs to be chained upto `trust_bundle` to be useful, while `auto_generated_edge_ca_expiry_days` is independent and mutually exclusive of both super-config values.

Reply:
```rust
#[serde(untagged)]
enum EdgeCa {
    Explicit {
        cert: Url,
        pk: Url,
    },
    QuickStart {
        auto_generated_expiry_days: u16,
    },
}
```

Reply:
Two questions:
- Are you suggesting this for super-config only? Or do we want to keep an `[edge_ca]` section in edged.toml
- Will `Explicit` enum value contain `trust_bundle_cert` too then?

Reply:
I've only been talking about the super-config.

---

If I personally had to choose whether `trust_bundle_cert` is in `[edge_ca]` or not, I would choose not to put it there. I only mentioned Varun and Micah asking for it. I believe their argument is that the only time the user cares about the `trust_bundle_cert` is when they're setting it up with nested Edge, in which case it'd be nice to have it bear the other property they have to set up for nested Edge - the Edge CA.

(In 1.1 there was a reason to colocate them even for non-nested Edge - the user had to make sure the trust bundle cert includes the device CA. This is no longer needed for 1.2, so apparently only the nested Edge use case remains.)

Reply:
Ok. I have replaced `EdgeCa` struct with the suggested enum. Thanks.